main.go

```diff
@@ -1,318 +1,65 @@
+		err := cmd.RunServer()
+		if err != nil {
+			log.Fatal(err)
+		}
 	}
 
 	if *shouldRunScheduleScenarios {
-		usecases := NewUseCases(appContext, appConfig, nil)
-		err := jobs.ScheduleDueScenarios(appContext, usecases, tracingConfig)
+		err := cmd.RunScheduleScenarios()
 		if err != nil {
-			logger.ErrorContext(appContext, "jobs.ScheduleDueScenarios failed", slog.String("error", err.Error()))
-			os.Exit(1)
-			return
+			log.Fatal(err)
 		}
 	}
 
 	if *shouldRunExecuteScheduledScenarios {
-		usecases := NewUseCases(appContext, appConfig, nil)
-		err := jobs.ExecuteAllScheduledScenarios(appContext, usecases, tracingConfig)
+		err := cmd.RunScheduledExecuter()
 		if err != nil {
-			logger.ErrorContext(appContext, "jobs.ExecuteAllScheduledScenarios failed", slog.String("error", err.Error()))
-			os.Exit(1)
-			return
+			log.Fatal(err)
 		}
 	}
 
 	if *shouldRunDataIngestion {
-		usecases := NewUseCases(appContext, appConfig, nil)
-		err := jobs.IngestDataFromCsv(appContext, usecases, tracingConfig)
+		err := cmd.RunBatchIngestion()
 		if err != nil {
-			logger.ErrorContext(appContext, "jobs.IngestDataFromCsv failed", slog.String("error", err.Error()))
-			os.Exit(1)
-			return
+			log.Fatal(err)
 		}
 	}
 
 	if *shouldRunScheduler {
-		jobs.RunScheduler(appContext, NewUseCases(appContext, appConfig, nil), tracingConfig)
-	}
-}
-
-func NewUseCases(ctx context.Context, appConfiguration AppConfiguration, marbleJwtSigningKey *rsa.PrivateKey) usecases.Usecases {
-	marbleConnectionPool, err := infra.NewPostgresConnectionPool(
-		appConfiguration.pgConfig.GetConnectionString())
-	if err != nil {
-		log.Fatal("error creating postgres connection to marble database", err.Error())
-	}
-
-	repositories, err := repositories.NewRepositories(
-		marbleJwtSigningKey,
-		infra.InitializeFirebase(ctx),
-		marbleConnectionPool,
-		infra.InitializeMetabase(appConfiguration.metabase),
-		appConfiguration.config.GcsTransferCheckEnrichmentBucket,
-	)
-	if err != nil {
-		slog.Error("repositories.NewRepositories failed", slog.String("error", err.Error()))
-		os.Exit(1)
-	}
-
-	return usecases.Usecases{
-		Repositories:  *repositories,
-		Configuration: appConfiguration.config,
-	}
-}
-
-func setupSentry(conf AppConfiguration) {
-	if err := sentry.Init(sentry.ClientOptions{
-		Dsn:           conf.sentryDsn,
-		EnableTracing: true,
-		Environment:   conf.env,
-		TracesSampler: sentry.TracesSampler(func(ctx sentry.SamplingContext) float64 {
-			if ctx.Span.Name == "GET /liveness" {
-				return 0.0
-			}
-			if ctx.Span.Name == "POST /ingestion/:object_type" {
-				return 0.05
-			}
-			if ctx.Span.Name == "POST /decisions" {
-				return 0.05
-			}
-			if ctx.Span.Name == "GET /token" {
-				return 0.05
-			}
-			return 0.1
-		}),
-		// Experimental - value to be adjusted in prod once volumes go up - relative to the trace sampling rate
-		ProfilesSampleRate: 0.2,
-	}); err != nil {
-		panic(err)
+		cmd.RunJobScheduler()
```

Comment:
Probably not necessary because it's the last one but just in case 
```suggestion
		err := cmd.RunJobScheduler()
		if err != nil {
			log.Fatal(err)
		}
```